File: programmers/전력망 둘로 나누기/solve.js
function countNode(tree, n, start, except) {
    const visit = Array(n + 1).fill(false);
    let   count = 0;
    
    visit[start] = true;
    function dfs(tree, node, visit, n) {
        for(let i = 1; i <= n; ++i) {
            const nodes = tree[node][i];
            if (visit[i]|| !nodes || (node === start && i === except))
                continue;
            visit[i] = true;
            dfs(tree, i, visit, n);
        }
    }
    dfs(tree, start, visit, n);
    for(const node of visit) {
        if(node)
            ++count;
    }
    // console.log(start, except, "Visit= ", visit, count);
    return count;
}

function solution(n, wires) {
    const tree = Array.from(Array(n + 1), () => Array(n + 1).fill(false) );
    for(const wire of wires) {
        tree[wire[0]][wire[1]] = true;
        tree[wire[1]][wire[0]] = true;
    }
    let answer = n;
    for(const wire of wires) {
        answer = Math.min(Math.abs(n - 2 * countNode(tree, n, wire[0],wire[1])), answer);
    }
    return answer;
}

File: programmers/등굣길/solve.cpp
#include <bits/stdc++.h>

using namespace std;

int solution(int m, int n, vector<vector<int>> puddles) {
    int map[101][101] = {0,};
    int answer = 0;
    
    for(int i = 0; i < puddles.size(); ++i)
        map[puddles[i][1]][puddles[i][0]] = -1;
    map[1][1] = 1;
    for(int y = 1; y <= n; ++y) {
        for(int x = 1; x <= m; ++x) {
            if(map[y][x] == -1) continue;
            map[y][x] += map[y - 1][x] == -1 ? 0 : map[y - 1][x];
            map[y][x] += map[y][x - 1] == -1 ? 0 : map[y][x - 1];
            map[y][x] %= (int)(1e9 + 7);
        }
    }
    
    return map[n][m];
}